R/hash_names.R

```diff
@@ -30,9 +32,15 @@
 #'   \code{data.frame}, including original labels, shortened hash, and full
 #'   hash.
 #'
+#' @param hashfun a hashing function that takes and returns a [raw][base::raw] vector of
```

Comment:
While it is nice to be able to specify the hashing function directly, this will be a bit above what basic users know how to do. A shorthand argument should be added for this, e.g. `method = xxx` where `xxx` can be `'fast'` or `'secure'`, pointing respectively to `sodium::scrypt` and something else; also would be good to check how much fast `sodium::sha256` is, and / or if there are some event faster alternatives. Bottomline is, for the *fast* option we only care about avoiding name collision when there are a few 100K entries, not about security.

Reply:
Gotcha. 

The speed difference is several orders of magnitude. I've added the fast/secure options along with the ability to specify the function.